pkg/apis/core/validation/validation.go

```diff
@@ -5136,8 +5136,9 @@ func ValidatePodUpdate(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 		return allErrs
 	}
 
-	if qos.GetPodQOS(oldPod) != qos.ComputePodQOS(newPod) {
-		allErrs = append(allErrs, field.Invalid(fldPath, newPod.Status.QOSClass, "Pod QoS is immutable"))
+	computedNewQOS := qos.ComputePodQOS(newPod)
+	if qos.GetPodQOS(oldPod) != computedNewQOS {
+		allErrs = append(allErrs, field.Invalid(fldPath, computedNewQOS, "Pod QoS is immutable"))
```

Comment:
Is it possible/easy to understand if there's an in-place scaling taking place? If it is, the problem can be made much clearer if we mention this was caused due to a resize IMO. WDYT?

Reply:
Yes, I think the only time this message can come out is when there is an in-place scaling taking place because:
-  the function we're in ( `ValidatePodUpdate`)  seems to only validate the spec, while the QoS is in the status 
-  we're comparing "old" QoS to "computed new" QoS, so "new"  QoS is ignored

So I think at least here it's just kind of quietly ignoring the case where you directly change the QoS. 

So hmm, keeping with the tone of the other validation errors, maybe something like `pod resource changes may not change pod QoS, pod QoS is immutable` ?